crates/storage/provider/src/traits/block.rs

```diff
@@ -12,6 +12,16 @@ use reth_primitives::{
 };
 use std::ops::RangeInclusive;
 
+/// Enum to control transaction hash inclusion.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, Hash, Default)]
+pub enum TransactionKind {
```

Comment:
I realized we already have a type called `TransactionKind` in primitives,

let's call this `TransactionVariant` because eventually, we could integrate this with #5098

because the fact that we return a fake hash here in the first place is not really ideal

Reply:
@mattsse Yes I just realized the same thing while searching the entire codebase, I just did the replacement to avoid confusion.